### PR TITLE
Field: move `pow` and `invert` to the trait def.

### PR DESCRIPTION
**Pull request by akoshelev**
These implementations are generic enough to be defined for any field, but they require some additional bounds on `Self::Integer` to be able to subtract, right shift and determine the size in bits of the underlying type

### CODE REVIEW

File: src/field.rs [hunk 1]
```diff
@@ -1,8 +1,30 @@
...
+// Trait for primitive integer types used to represent the underlying type for field values
+pub trait Int:
+    Sized
+    + Copy
+    + Debug
+    + Ord
+    + Sub<Output = Self>
+    + Into<u128>
+    + Shr<u32, Output = Self>
+    + BitAnd<u8, Output = Self>
```
Comment (martinthomson) [inline src/field.rs:16]: Doesn't this need to be:

```suggestion
    + BitAnd<Self, Output = Self>
```

Comment (martinthomson) [reply src/field.rs:16]: I guess then you would need to define `const ONE: Self` as well.

You could bound with `From<u8>` and drop both `ZERO` and `ONE`.

I'm mostly just concerned that this won't work when the Field uses a different integer basis.

**Review by martinthomson — APPROVED** [review]
This looks like it might work pretty well.  I like the general shape of it.

File: src/field.rs (hunk 1 above)
Comment (akoshelev) [reply src/field.rs]: yea I was thinking about it too - don't have a good explanation why I decided against it. It should always be safe to upcast from `u8`.

**Comment by thurstonsand** [comment]
LGTM

**Pull request merged**